syserol/figures/figure2.py

```diff
@@ -0,0 +1,57 @@
+import numpy as np
+import seaborn as sns
+from .common import getSetup, subplotLabel
+from syserol.COVID import Tensor4D, dayLabels, dimensionLabel3D, pbsSubtractOriginal
+from syserol.tensor import perform_CMTF, tensor_degFreedom, cp_normalize, reorient_factors, sort_factors
+from matplotlib.ticker import ScalarFormatter
+from itertools import groupby
+
+
+
+def makeFigure():
+    ax, f = getSetup((13, 9), (1, 4))
+
+    comps = np.arange(1,3)
+    tensor, _ = Tensor4D()
+    CMTFfacs = [perform_CMTF(tensor, cc) for cc in comps]
+
+    CMTFfacs = [cp_normalize(f) for f in CMTFfacs]
+    CMTFfacs = [reorient_factors(f) for f in CMTFfacs]
+    CMTFfacs = [sort_factors(f) if i > 0 else f for i,
+                f in enumerate(CMTFfacs)]
+
+    Rlabels, agLabels = dimensionLabel3D()
+    days = dayLabels()
+    tfac = CMTFfacs[1]
+
+    df = pbsSubtractOriginal()
+    components = [str(ii + 1) for ii in range(tfac.rank)]
+    comp_plot(tfac.factors[0], components,
+              list(df.loc[np.unique(df['patient_ID'])]['group']), "Subjects", ax[0], True)
+    comp_plot(tfac.factors[1], components, agLabels, "Antigens", ax[1])
+    comp_plot(tfac.factors[2], components, Rlabels, "Receptors", ax[2])
+    comp_plot(tfac.factors[3], components, days, "Time (days)", ax[3])
```

Comment:
Don't have to do it on this PR, but I think the time would be visualized more effectively as a line plot. For example, see Fig. 2F here: https://www.biorxiv.org/content/10.1101/2021.07.03.451002v1

Reply:
So in that figure, some of them don't look exactly unimodal? For example, component 1 goes up and then back down?

Reply:
Right. That's not using your code. The visualization is still useful as an example.

Reply:
I see, I thought maybe because it was also time it would also have similar behavior.